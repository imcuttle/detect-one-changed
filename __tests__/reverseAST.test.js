/**
 * @file reverseAST
 * @author Cuttle Cong
 * @date 2018/11/2
 * @description
 */

const remark = require('remark')
const reverseAST = require('../lib/reverseAST')

describe('reverseAST', function() {
  it('should reverseAST', () => {
    let input = [
      '# Root\n',
      '- Father',
      '  - Bob',
      '    - July',
      '  - John',
      '  - Kiyi\n',
      '- Mother',
      '  - Bob',
      '    - July',
      '    - Kil',
      '  - John',
      '  - Kiyi'
    ].join('\n')
    const ast = remark()
      .data('settings', { position: false })
      .parse(input)

    input = remark().stringify(ast)

    reverseAST(ast)

    expect(ast).toMatchSnapshot()

    expect(remark().stringify(reverseAST(ast))).toBe(input)
  })
})
