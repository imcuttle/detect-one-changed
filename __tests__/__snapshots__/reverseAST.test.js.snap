// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`reverseAST should reverseAST 1`] = `
Object {
  "children": Array [
    Object {
      "children": Array [
        Object {
          "checked": null,
          "children": Array [
            Object {
              "children": Array [
                Object {
                  "checked": null,
                  "children": Array [
                    Object {
                      "children": Array [
                        Object {
                          "type": "text",
                          "value": "Kiyi",
                        },
                      ],
                      "type": "paragraph",
                    },
                  ],
                  "spread": false,
                  "type": "listItem",
                },
                Object {
                  "checked": null,
                  "children": Array [
                    Object {
                      "children": Array [
                        Object {
                          "type": "text",
                          "value": "John",
                        },
                      ],
                      "type": "paragraph",
                    },
                  ],
                  "spread": false,
                  "type": "listItem",
                },
                Object {
                  "checked": null,
                  "children": Array [
                    Object {
                      "children": Array [
                        Object {
                          "checked": null,
                          "children": Array [
                            Object {
                              "children": Array [
                                Object {
                                  "type": "text",
                                  "value": "Kil",
                                },
                              ],
                              "type": "paragraph",
                            },
                          ],
                          "spread": false,
                          "type": "listItem",
                        },
                        Object {
                          "checked": null,
                          "children": Array [
                            Object {
                              "children": Array [
                                Object {
                                  "type": "text",
                                  "value": "July",
                                },
                              ],
                              "type": "paragraph",
                            },
                          ],
                          "spread": false,
                          "type": "listItem",
                        },
                      ],
                      "ordered": false,
                      "spread": false,
                      "start": null,
                      "type": "list",
                    },
                    Object {
                      "children": Array [
                        Object {
                          "type": "text",
                          "value": "Bob",
                        },
                      ],
                      "type": "paragraph",
                    },
                  ],
                  "spread": false,
                  "type": "listItem",
                },
              ],
              "ordered": false,
              "spread": false,
              "start": null,
              "type": "list",
            },
            Object {
              "children": Array [
                Object {
                  "type": "text",
                  "value": "Mother",
                },
              ],
              "type": "paragraph",
            },
          ],
          "spread": false,
          "type": "listItem",
        },
        Object {
          "checked": null,
          "children": Array [
            Object {
              "children": Array [
                Object {
                  "checked": null,
                  "children": Array [
                    Object {
                      "children": Array [
                        Object {
                          "type": "text",
                          "value": "Kiyi",
                        },
                      ],
                      "type": "paragraph",
                    },
                  ],
                  "spread": false,
                  "type": "listItem",
                },
                Object {
                  "checked": null,
                  "children": Array [
                    Object {
                      "children": Array [
                        Object {
                          "type": "text",
                          "value": "John",
                        },
                      ],
                      "type": "paragraph",
                    },
                  ],
                  "spread": false,
                  "type": "listItem",
                },
                Object {
                  "checked": null,
                  "children": Array [
                    Object {
                      "children": Array [
                        Object {
                          "checked": null,
                          "children": Array [
                            Object {
                              "children": Array [
                                Object {
                                  "type": "text",
                                  "value": "July",
                                },
                              ],
                              "type": "paragraph",
                            },
                          ],
                          "spread": false,
                          "type": "listItem",
                        },
                      ],
                      "ordered": false,
                      "spread": false,
                      "start": null,
                      "type": "list",
                    },
                    Object {
                      "children": Array [
                        Object {
                          "type": "text",
                          "value": "Bob",
                        },
                      ],
                      "type": "paragraph",
                    },
                  ],
                  "spread": false,
                  "type": "listItem",
                },
              ],
              "ordered": false,
              "spread": false,
              "start": null,
              "type": "list",
            },
            Object {
              "children": Array [
                Object {
                  "type": "text",
                  "value": "Father",
                },
              ],
              "type": "paragraph",
            },
          ],
          "spread": false,
          "type": "listItem",
        },
      ],
      "ordered": false,
      "spread": true,
      "start": null,
      "type": "list",
    },
    Object {
      "children": Array [
        Object {
          "type": "text",
          "value": "Root",
        },
      ],
      "depth": 1,
      "type": "heading",
    },
  ],
  "type": "root",
}
`;
